app/decorators/room_decorator.rb

```diff
@@ -17,8 +17,13 @@ def access_control_options_for_select
     end
   end
 
-  def badge_color_for_status
-    BADGE_COLORS[status.to_sym]
+  def status_to_component
+    return unless Room.statuses.key?(status)
```

Comment:
```suggestion
    return { plain: nil } unless Room.statuses.key?(status)
```

Otherwise it may fail the render in view